jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java

```diff
@@ -229,24 +224,21 @@ private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _pack
 
                 int index = fqn.lastIndexOf(".") + 1;
                 if (index == 0) { //Actually not a fully qualified name
-                    fqn = _package.name() + "." + fqn;
+                    fqn = container.getPackage().name() + "." + fqn;
                     index = fqn.lastIndexOf(".") + 1;
                 }
 
-                if (index >= 0 && index < fqn.length()) {
+                if (index < fqn.length()) {
                     fqn = fqn.substring(0, index) + ruleFactory.getGenerationConfig().getClassNamePrefix() + fqn.substring(index) + ruleFactory.getGenerationConfig().getClassNameSuffix();
                 }
 
-                if (usePolymorphicDeserialization) {
```

Comment:
It seems as this branch had no difference in outcome from the other one, hence it was removed.